scripts/process_uploads.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+"""
+Processes files uploaded to gcp
+"""
+
+import os
+import csv
+from os.path import join, basename
+import subprocess
+import tempfile
+import shutil
+from typing import List
+import hail as hl
+import hailtop.batch as hb
+from joint_calling.upload_processor import batch_move_files
+
+
+def samples_from_csv(local_csv_path):
+    """ Determines list of samples from a given csv file. """
+
+    samples: List[str] = []
+    with open(local_csv_path) as csv_file:
+        csv_reader = csv.DictReader(csv_file)
+        for row in csv_reader:
+            samples.append(row['sample.sample_name'])
+
+    return samples
+
+
+def determine_samples(upload_bucket: str, previous_batch_path: str):
+    """Determine files that should be moved to main vs archive bucket.
+    Determines the difference between the latest CSV file within upload
+    and the CSV file from the most recent batch as the list of samples to
+    be moved.
+
+    Assumptions
+    ==========
+    Only one .csv file exists in the upload bucket and previous batch directory.
+    """
+    local_tmp_dir = tempfile.mkdtemp()
+    all_samples: List[str] = []
+    previous_samples: List[str] = []
+
+    # Pull the samples listed in the most recent CSV file
+    cmd = f'gsutil ls \'gs://{upload_bucket}/*.csv\''
+    curr_csv_file_path = subprocess.check_output(cmd, shell=True).decode().strip()
+    local_curr_csv_path = join(local_tmp_dir, basename(curr_csv_file_path))
+    subprocess.run(
+        f'gsutil cp {curr_csv_file_path} {local_curr_csv_path}', check=False, shell=True
+    )
+
+    all_samples = samples_from_csv(local_curr_csv_path)
```

Comment:
Similarly here you don't need to copy the file to a local directory. You can either use the blob's `file`-like API or download the string contents and use an approach like https://stackoverflow.com/questions/31658115/python-csv-dictreader-parse-string/31658188 for the `csv.DictReader`.

Reply:
Done!